code/game/atoms.dm

```diff
@@ -642,15 +642,17 @@ var/list/blood_splatter_icons = list()
 	update_icons()	//apply the now updated overlays to the mob
 
 
-/atom/proc/add_vomit_floor(mob/living/carbon/M as mob, var/toxvomit = 0)
-	if( istype(src, /turf/simulated) )
-		var/obj/effect/decal/cleanable/vomit/this = new /obj/effect/decal/cleanable/vomit(src)
+/atom/proc/add_vomit_floor(toxvomit = 0, type = /obj/effect/decal/cleanable/vomit)
+	playsound(src, 'sound/effects/splat.ogg', 50, 1)
+	if(istype(src, /turf/simulated))

```

Comment:
There's really no need for this to exclude simulated turfs, to be honest. I really have no idea why it has after all this time.